kernel/common/enumerator.rb

```diff
@@ -148,7 +149,19 @@ def rewind
     end
 
     def size
-      @size.kind_of?(Proc) ? @size.call : @size
+      if @size.kind_of?(Proc)
+        @size.call
+      elsif @size.is_a? Numeric
```

Comment:
As a minor style note, I'd suggest using `kind_of?(Numeric)` here be consistent with `kind_of?(Proc)`